dashboard/app/controllers/api/v1/regional_partners_controller.rb

```diff
@@ -5,4 +5,25 @@ def index
 
     render json: regional_partners.order(:name).map {|partner| {id: partner.id, name: partner.name}}
   end
+
+  # GET /api/v1/regional_partners/capacity?role=:role&regional_partner_filter=:regional_partner
+  def capacity
+    role = params[:role]
+    regional_partner_filter = current_user.workshop_admin? ? params[:regional_partner_filter] : current_user.regional_partners.first.try(:id)
+
+    render json: {capacity: get_partner_cohort_capacity(regional_partner_filter, role)}
+  end
+
+  def get_partner_cohort_capacity(regional_partner_filter, role)
+    unless ['none', 'all'].include? regional_partner_filter
+      partner_id = regional_partner_filter ? regional_partner_filter : current_user.regional_partners.first
+      regional_partner = RegionalPartner.find_by(id: partner_id)
+      if role == 'csd_teachers'
+        return regional_partner.cohort_capacity_csd
+      elsif role == 'csp_teachers'
+        return regional_partner.cohort_capacity_csp
+      end
+    end
```

Comment:
We want this to return `nil` for the `all` filter? Since this is not an action, mark this method private or protected and add a header comment w a short description, `@param`s and `@return`.